leader/leader_test.go

```diff
@@ -0,0 +1,187 @@
+package leader
+
+import (
+	"context"
+	"os"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+var _ = Describe("Leader", func() {
+
+	Describe("Become", func() {
+		var (
+			client crclient.Client
+		)
+		BeforeEach(func() {
+			client = fake.NewFakeClient(
+				&corev1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "leader-test",
+						Namespace: "testns",
+						OwnerReferences: []metav1.OwnerReference{
+							{
+								APIVersion: "v1",
+								Kind:       "Pod",
+								Name:       "leader-test",
+							},
+						},
+					},
+				},
+				&corev1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "leader-test",
+						Namespace: "testns",
+						OwnerReferences: []metav1.OwnerReference{
+							{
+								APIVersion: "v1",
+								Kind:       "Pod",
+								Name:       "leader-test",
+							},
+						},
+					},
+				},
+			)
+		})
+		It("should return an error when POD_NAME is not set", func() {
+			os.Unsetenv("POD_NAME")
+			err := Become(context.TODO(), "leader-test")
+			Expect(err).ShouldNot(BeNil())
+		})
+		It("should not return an error", func() {
+			os.Setenv("POD_NAME", "leader-test")
+			readNamespace = func() ([]byte, error) {
+				return []byte("testns"), nil
+			}
+
+			err := Become(context.TODO(), "leader-test", WithClient(client))
+			Expect(err).Should(BeNil())
+		})
```

Comment:
Can we add a test case for returning `ErrNoNamespace` from `Become` and using `errors.Is(err, ErrNoNamespace)`?